enhancements/security/bmc-ca-certificate-support.md

```diff
@@ -0,0 +1,288 @@
+  if it does, then:
+  - Write the CA certificates mounted in `/certs/ca/bmc` to a file, e.g.
+    `/tmp/bmc-tls.pem`.
+  - Launch a process to monitor changes to `/certs/ca/bmc`, if changes are
+    detected, then update `/tmp/bmc-tls.pem` accordingly.
+  - Set the `default_verify_ca_path` in `ironic.conf` using `/tmp/bmc-tls.pem`
+    as the value.
+
+- Modify [installer][OpenShift Installer] for control plane installation:
+  - Add a new optional field `platform.baremetal.bmcCACert` in
+    `install-config.yaml` to allow users to enter the contents of the CA
+    certificates.
+  - Perform a pre-validation of the SSL connection with the BMCs using the
+    provided CA certificates before the installation begins. If the validation
+    fails, an error message will be outputed to indicate which baremetal
+    failed. This helps to identify issues early on, especially for the
+    time-consuming IPI installation.
+  - Modify `startironic.sh.template`:
+    - Create the CA certificate files in of the bootstrap VM
+      according to `platform.baremetal.bmcCACert`.
+    - Mount the CA certificate files when starting the Ironic container.
+    - Create the contents of CA certificate files as a ConfigMap.
+
+- Modify [cluster-baremetal-operator][OpenShift Cluster Baremetal Operator]
+  to ensure mounting the `bmc-verify-ca` ConfigMap into Ironic container
+  for worker node deployment, and unmounting it when `bmc-verify-ca`
+  ConfigMap is deleted.
+
+### Workflow Description
+
+**cluster creator** is a human user responsible for deploying a cluster.
+
+**cluster administrator** is a human user responsible for managing a cluster.
+
+- Deploy a cluster via IPI installation:
+  1. The cluster creator enters the contents of the CA certificate under the
+     `platform.baremetal.bmcCACert` field in `install-config.yaml`.
+  2. The cluster creator ensures that `disableCertificateVerification`
+     is not set to true in `install-config.yaml`.
+  3. The cluster creator executes the installer command to install the cluster.
+
+- Update the CA certificate for an existing cluster:
+  1. The cluster administrator modifies the contents of the `bmc-verify-ca`
+     ConfigMap in `openshift-machine-api` namespace and applies the changes.
+
+- Enable secure SSL communication with BMCs for an existing cluster without
+  initial CA certificate configuration:
+  1. The cluster administrator creates a ConfigMap named `bmc-verify-ca` in
+     `openshift-machine-api` namespace, containing the desired CA certificates.
+  2. The cluster administrator confirms that the `bmc-verify-ca` ConfigMap has
+     been mounted to `/certs/ca/bmc` in Ironic container.
+  3. The cluster administrator ensures that `disableCertificateVerification`
+     is not set to true in BareMetalHost objects.
+
+### API Extensions
+
+Add an optional map field `bmcCACert` under `platform.baremetal` in
+`install-config.yaml`. Users can configure multiple sets of CA certificates
```

Comment:
Can't they do that by just `cat`ing them all together in a single field?

Reply:
As the install-config.yaml example below, this is just a single field.
Using map format allows cluster administrators to assign distinct names to their CA certs, allowing administrators to update them easily in the subsequent CA cert ConfigMap.

Reply:
But that's a different format from `additionalTrustBundle`, right? This difference can be a source of confusion for operators.